File: files/CH-074.py
import pandas as pd

# Read the Excel file
path = 'CH-074 Determining missing fields.xlsx'
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=14)
test = pd.read_excel(path, usecols="F:H", skiprows=1, nrows=4)

input['cumsum'] = (input['Info'] == "Name").cumsum()
result = (input.groupby(['cumsum', 'Info'])
          .size()
          .unstack(fill_value=0)
          .reset_index()
          .melt(id_vars=['cumsum'], var_name='Info', value_name='n')
          .groupby(['cumsum', 'n'])['Info']
          .apply(list)
          .unstack(fill_value=[])
          .reset_index())
result['Missing fields'] = result[0].apply(lambda x: "-" if not x else ", ".join(x))
result['Duplicate Fields'] = result[2].apply(lambda x: "-" if not x else ", ".join(x))
result = result[['cumsum', 'Missing fields', 'Duplicate Fields']]
result.columns = ['Record No', 'Missing fields', 'Duplicate Fields']

print(result)
print(test)

# One field differs, because there is empty cell where hyphen should be in H3

File: files/000-099/074/CH-074.R
library(tidyverse)
library(readxl)

path = 'files/CH-074 Determining missing fields.xlsx'
input = read_excel(path, range = "B2:C15")
test  = read_excel(path, range = "F2:H6")

result = input %>%
  mutate(cumsum = cumsum(`Info...1` == "Name")) %>%
  select(-Info...2) %>% 
  summarise(n = n(), .by = c(cumsum, `Info...1`)) %>%
  pivot_wider(names_from = `Info...1`, values_from = n, values_fill = list(n = 0)) %>%
  pivot_longer(cols = -cumsum, names_to = "Info", values_to = "n") %>%
  pivot_wider(names_from = n, values_from = Info, values_fn = list) %>%
  select(`Record No` = 1, `Missing fields` = 4, `Duplicated Fields` = 3) %>%
  mutate(`Missing fields` = map_chr(`Missing fields`, ~ifelse(is.null(.x), "-", paste(.x, collapse = ", "))),
         `Duplicated Fields` = map_chr(`Duplicated Fields`, ~ifelse(is.null(.x), "-", paste(.x, collapse = ", "))),
         `Record No` = as.numeric(`Record No`))
         
print(result)
print(test)
# One field differs, because there is empty cell where hyphen should be in H3